Source/CarthageKit/Simulator.swift

```diff
@@ -18,8 +18,14 @@ internal struct Simulator: Decodable {
 		// Since Xcode 10.1, `availability` field is obsolated.
 		// Using `isAvailable` instead. its value is possible to be `YES` or `NO`.
 		let availability = try container.decodeIfPresent(String.self, forKey: .availability)
-		let isAvailable = try container.decodeIfPresent(String.self, forKey: .isAvailable)
-		self.isAvailable = isAvailable == "YES" || availability == "(available)"
+		
+		do {
+			let isAvailable = try container.decodeIfPresent(String.self, forKey: .isAvailable)
+			self.isAvailable = isAvailable == "YES" || availability == "(available)"
+		} catch {
+			// Xcode 10.1 uses key `isAvailable` with bool value so try to keep it backward compatible
+			self.isAvailable = try container.decode(Bool.self, forKey: .isAvailable)
```

Comment:
I prefer to use optional binding rather than exceptions.

```swift
if let isAvailable = try container.decodeIfPresent(Bool.self, forKey: .isAvailable) {
    // Xcode 10.1 ~
    self.isAvailable = isAvailable
if let availability = try container.decodeIfPresent(String.self, forKey: .availability), availability == "(available)" {
    // <= Xcode 10.0
    self.isAvailable = true
} else if isAvailable = try container.decodeIfPresent(String.self, forKey: .availability), isAvailable == "YES" {
    // Xcode 10.1 beta
    self.isAvailable = true
}
```

Reply:
Well the idea of having this case as the last one was to be able to take advantage of `Codable` exceptions in case when `isAvailable` key would be missing completely so I don't have to deal with default values as `isAvailable` property is required in `Simulator` model.